spdx/parsers/rdfbuilders.py

```diff
@@ -184,19 +186,23 @@ class PackageBuilder(tagvaluebuilders.PackageBuilder):
     def __init__(self):
         super(PackageBuilder, self).__init__()
 
-    def set_pkg_chk_sum(self, doc, chk_sum):
+    def set_pkg_checksum(self, doc, checksum: [Algorithm, Dict]):
         """
-        Set the package check sum, if not already set.
-        chk_sum - A string
-        Raise CardinalityError if already defined.
+        Set the package checksum.
+        checksum - A string
+        Raise SPDXValueError if checksum type invalid.
         Raise OrderError if no package previously defined.
         """
         self.assert_package_exists()
-        if not self.package_chk_sum_set:
-            self.package_chk_sum_set = True
-            doc.packages[-1].checksum = checksum.Algorithm("SHA1", chk_sum)
+        if isinstance(checksum, dict):
+            algo = checksum.get('algorithm') or 'SHA1'
+            if algo.startswith('checksumAlgorithm_'):
+                algo = convert_rdf_checksum_algorithm(algo) or 'SHA1'
```

Comment:
No, it isn't. Thanks to the inherited structure the `jsonyamlxml`-parser uses this method to parse the package checksum, so this could acutally be a dict. I added a check for `str` as this would also be a valid input from the `jsonyamlxml` parser, at least for backwards compatibility...

Reply:
Oh dear... alright then :man_facepalming: